includes/QueryPage.php

```diff
@@ -443,11 +443,44 @@ public function getCachedTimestamp() {
 	}
 
 	/**
+	 * Get max number of results we can return.
+	 *
+	 * Most QueryPage subclasses use $wgQueryCacheLimit as a default value, but this can be customized.
+	 *
+	 * @author macbre
+	 * @see SUS-1675
+	 *
+	 * @return int
+	 */
+	protected function getMaxResults() {
+		global $wgQueryCacheLimit, $wgQueryPages;
+
+		// that's our default return value
+		$limit = $wgQueryCacheLimit;
+
+		// get the $wgQueryPages entry for the current QueryPage class
+		$currentQueryPage = get_class( $this );
+		$entries = array_values( array_filter(
```

Comment:
`array_filter` keeps the key from the original array. `array_values` will make sure that we can easily use `[0]` to get the only item in `$entries`.